### PR TITLE
combineLatest

### PR DESCRIPTION
**Pull request by eulerfx**

### CODE REVIEW

File: src/FSharp.Control.AsyncSeq/AsyncSeq.fsi [hunk 1]
```diff
@@ -294,6 +294,16 @@ module AsyncSeq =
     /// Feeds an async sequence of values into an async sequence of functions.
     val zapp : functions:AsyncSeq<('T -> 'U)> -> source:AsyncSeq<'T> -> AsyncSeq<'U>
 
+    /// Merges two async sequences using the specified combine function. The resulting async sequence produces an element when either
+    /// input sequence produces an element, passing the new element from the emitting sequence and the previously emitted element from the other sequence.
+    /// If either of the input sequences is empty, the resulting sequence is empty.
+    val combineLatestAsync : combine:('T -> 'U -> Async<'V>) -> source1:AsyncSeq<'T> -> source2:AsyncSeq<'U> -> AsyncSeq<'V>
```
Comment (dsyme) [inline src/FSharp.Control.AsyncSeq/AsyncSeq.fsi]: We should probably start writing samples for each function in the docs.

File: tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs [hunk 2]
```diff
@@ -1219,4 +1224,19 @@ let ``AsyncSeq.groupBy should propagate exception and terminate all groups``() =
...
+  for n in 0..20 do
+    for m in 0..10 do
+      let ls1 = List.init n id
+      let ls2 = List.init m id 
+      // expect each element to increase combined sum by 1
+      // expected count is sum of source counts minus 1 for first result
+      let expectedCount = 
+        if n = 0 || m = 0 then 0
+        else max (n + m - 1) 0
+      let expected = List.init expectedCount id |> AsyncSeq.ofSeq
```
Comment (dsyme) [inline tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs]: Feels like we should add some tests where the inputs have edge-case timing behaviour?

File: src/FSharp.Control.AsyncSeq/AsyncSeq.fsi (hunk 1 above)
Comment (eulerfx) [inline src/FSharp.Control.AsyncSeq/AsyncSeq.fsi]: Agreed. Adding for this one shortly.

File: tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs (hunk 2 above)
Comment (eulerfx) [inline tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs]: I pushed an additional test to check for a timeout. Is this what you had in mind?

Comment (dsyme) [inline tests/FSharp.Control.AsyncSeq.Tests/AsyncSeqTests.fs]: Yes, great

**Comment by eulerfx** [comment]
Should this be renamed to `combineLatestWith` with `combineLatest` returning a pair?

File: docs/content/library/AsyncSeqExamples.fsx [hunk 3]
```diff
@@ -94,18 +98,84 @@ The above workflow:
...
+| source2 |    | b0 | b1 |    |   |    |
+| result  |    | c0 | c1 | c2 |   | c3 |
+----------------------------------------
+
+where
+
+c0 = f a0 b0
+c1 = f a0 b1
+c2 = f a1 b1
+c4 = f a2 b1
```
Comment (mexx) [inline docs/content/library/AsyncSeqExamples.fsx]: should be `c3`, right?

Comment (eulerfx) [inline docs/content/library/AsyncSeqExamples.fsx]: That's correct, thanks.
